### PR TITLE
[freerdp] Update to 3.3.0

### PR DESCRIPTION
**Pull request by dg0yt**

### CODE REVIEW

**Comment by dg0yt** [comment]
CI for all platforms: https://dev.azure.com/vcpkg/public/_build/results?buildId=100029

arm64-windows:
~~~
FAILED: freerdp3.dll freerdp3.lib 
cmd.exe /C "cd . && D:\downloads\tools\cmake-3.27.1-windows\cmake-3.27.1-windows-i386\bin\cmake.exe -E vs_link_dll --intdir=libfreerdp\CMakeFiles\freerdp.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MICROS~1\2022\ENTERP~1\VC\Tools\MSVC\1438~1.331\bin\Hostx64\arm64\link.exe  @CMakeFiles\freerdp.rsp  /out:freerdp3.dll /implib:freerdp3.lib /pdb:freerdp3.pdb /dll /version:3.3 /machine:ARM64 /nologo    /debug /INCREMENTAL  /INCREMENTAL:NO /LTCG  && cd ."
LINK: command "C:\PROGRA~1\MICROS~1\2022\ENTERP~1\VC\Tools\MSVC\1438~1.331\bin\Hostx64\arm64\link.exe @CMakeFiles\freerdp.rsp /out:freerdp3.dll /implib:freerdp3.lib /pdb:freerdp3.pdb /dll /version:3.3 /machine:ARM64 /nologo /debug /INCREMENTAL /INCREMENTAL:NO /LTCG /MANIFEST:EMBED,ID=2" failed (exit code 1120) with the following output:
   Creating library freerdp3.lib and object freerdp3.exp


rfx_rlgr.c.obj : error LNK2001: unresolved external symbol __lzcnt


freerdp3.dll : fatal error LNK1120: 1 unresolved externals
~~~

x64-uwp:
~~~
D:\b\freerdp\src\3.3.0-04735aa4a0.clean\winpr\include\winpr/environment.h(90): error C2371: 'GetEnvironmentStringsW': redefinition; different basic types
~~~

x64-windows:
~~~
FAILED: server/Windows/CMakeFiles/wfreerdp-server.dir/wf_rdpsnd.c.obj 
C:\PROGRA~1\MICROS~1\2022\ENTERP~1\VC\Tools\MSVC\1438~1.331\bin\Hostx64\x64\cl.exe   -DFREERDP_EXPORTS -DUNICODE -DWIN32_LEAN_AND_MEAN -DWINVER=0x0601 -DWITH_OPENSSL -DWITH_SMARTCARD_EMULATE -DWITH_VERBOSE_WINPR_ASSERT -D_CRT_NONSTDC_NO_DEPRECATE -D_CRT_SECURE_NO_WARNINGS -D_UNICODE -D_WIN32_WINNT=0x0601 -D_WINSOCK_DEPRECATED_NO_WARNINGS -Dwfreerdp_server_EXPORTS -ID:\installed\x64-windows\include -ID:\b\freerdp\src\3.3.0-04735aa4a0.clean\winpr\include -ID:\b\freerdp\x64-windows-dbg\winpr\include -ID:\b\freerdp\x64-windows-dbg -ID:\b\freerdp\x64-windows-dbg\include -ID:\b\freerdp\src\3.3.0-04735aa4a0.clean\include -ID:\b\freerdp\src\3.3.0-04735aa4a0.clean\server\Windows\. /nologo /DWIN32 /D_WINDOWS  /utf-8 /MP  /W4 /wo4324 /Gd /Zi /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  -std:c11 -MDd /GL /showIncludes /Foserver\Windows\CMakeFiles\wfreerdp-server.dir\wf_rdpsnd.c.obj /Fdserver\Windows\CMakeFiles\wfreerdp-server.dir\ /FS -c D:\b\freerdp\src\3.3.0-04735aa4a0.clean\server\Windows\wf_rdpsnd.c
cl : Command line warning D9025 : overriding '/Zi' with '/Z7'
D:\b\freerdp\src\3.3.0-04735aa4a0.clean\include\freerdp/settings_types_private.h(61): warning C4324: 'rdp_settings': structure was padded due to alignment specifier
D:\b\freerdp\src\3.3.0-04735aa4a0.clean\server\Windows\wf_rdpsnd.c(78): error C2065: 'i': undeclared identifier
~~~

**Comment by jimwang118** [comment]
Usage test pass with following triplets:

```
x64-linux
```

File: ports/freerdp/portfile.cmake [hunk 1]
```diff
@@ -2,17 +2,20 @@ vcpkg_from_github(
...
     PATCHES
         dependencies.patch
-        DontInstallSystemRuntimeLibs.patch
         install-layout.patch
         keep-dup-libs.patch
         windows-linkage.patch
         wfreerdp-server-cli.patch
+        wf-rdpsnd.patch
+        msvc-arm64.patch
+        mingw-tp.patch
```
Comment (BillyONeal) [inline ports/freerdp/portfile.cmake:15]: These 4 seem pretty universal and upstream seems responsive, can you submit to https://github.com/FreeRDP/FreeRDP ?

Thanks!

**Comment by BillyONeal** [comment]
Thanks for the update!

File: ports/freerdp/portfile.cmake (hunk 1 above)
Comment (dg0yt) [reply ports/freerdp/portfile.cmake:15]: The first is already done upstream. The others are there now: https://github.com/FreeRDP/FreeRDP/pull/9964, https://github.com/FreeRDP/FreeRDP/pull/9965

**Pull request merged**